commands/01-mod.js

```diff
@@ -158,18 +158,37 @@ const Module = new Augur.Module()
           if (bannerHighRole.comparePositionTo(bannedHighRole) <= 0) {
             msg.reply(`you cannot ban ${u.escapeText(member.displayName)}!`);
             continue;
-          } else {
+          }
+          members.set(userId, member);
+        }
+        else {
+          members.set(userId, undefined);
+        }
+      }
+    }
+    if (members.size == 0) {
+      msg.reply("you need to tell me who to ban!").then(u.clean);
+      return;
+    }
+
+    let confirm = await u.confirm(msg, `Are you sure you want to ban the following?\n${members.keyArray().map(m => members.get(m) ? u.escapeText(members.get(m).displayName) : m).join("\n")}`);
+
+    if (confirm) {
+      let banCount = 0;
+      for (const [memberId, member] of members) {
+        try {
+          if (member) {
             const infraction = {
               discordId: member.id,
               description: reason,
               value: 30,
               mod: msg.author.id
             };
-            await (member.send(`You were banned from ${msg.guild.name} for ${reason}`).catch(() => blocked(member)));
+            let inf = await Module.db.infraction.save(infraction);
 
+            await (member.send(`You were banned from ${msg.guild.name} for ${reason}`).catch(() => blocked(member)));
             if (!msg.client.ignoreNotifications) msg.client.ignoreNotifications = new Set();
             msg.client.ignoreNotifications.add(member.id);
-
             await member.ban({days: 2, reason});
```

Comment:
Safe method
```suggestion
            await msg.guild.members.ban(MemberId, {days: 2, reason});
```

Reply:
This suggestion has a capitalization issue.

Reply:
Let's see if I can use a comment to resolve this while I'm already on this page...

```suggestion
            await member.ban({days: 2, reason});
            await msg.guild.members.ban(memberId, {days: 2, reason});
```